src/hotspot/cpu/x86/sharedRuntime_x86.cpp

```diff
@@ -83,3 +84,72 @@ void SharedRuntime::inline_check_hashcode_from_object_header(MacroAssembler* mas
 }
 #endif //COMPILER1
 
+#ifdef _WIN64
+const juint  float_sign_mask  = 0x7FFFFFFF;
+const juint  float_infinity   = 0x7F800000;
+const julong double_sign_mask = CONST64(0x7FFFFFFFFFFFFFFF);
+const julong double_infinity  = CONST64(0x7FF0000000000000);
+#endif
+
+#if !defined(_WINDOWS) || defined(_WIN64)
+JRT_LEAF(jfloat, SharedRuntime::frem(jfloat x, jfloat y))
+// see SharedRuntime::drem
+#ifdef _WIN64
+  // 64-bit Windows on amd64 returns the wrong values for
+  // infinity operands.
+  union { jfloat f; juint i; } xbits, ybits;
+  xbits.f = x;
+  ybits.f = y;
+  // x Mod Infinity == x unless x is infinity
+  if (((xbits.i & float_sign_mask) != float_infinity) &&
+       ((ybits.i & float_sign_mask) == float_infinity) ) {
+    return x;
+  }
+  return ((jfloat)fmod_winx64((double)x, (double)y));
+#else
+  jfloat retval;
+  asm ("\
+1:               \n\
+fprem            \n\
+fnstsw %%ax      \n\
+test   $0x4,%%ah \n\
+jne    1b        \n\
+"
+    :"=t"(retval)
+    :"0"(x), "u"(y)
+    :"cc", "ax");
+  return retval;
+#endif
+JRT_END
+
+JRT_LEAF(jdouble, SharedRuntime::drem(jdouble x, jdouble y))
+// Java bytecode drem is defined as C fmod (not C drem==remainder).
+// GCC had slow fmod():
+// since https://gcc.gnu.org/git/gitweb.cgi?p=gcc.git;h=4f2611b6e872c40e0bf4da38ff05df8c8fe0ee64
+// until https://gcc.gnu.org/git/gitweb.cgi?p=gcc.git;h=8020c9c42349f51f75239b9d35a2be41848a97bd
```

Comment:
We typically list such URLs in the bug database rather than the source code.

Reply:
I have removed it (but I do not like it much). Also removed it from the testcase.